coordinator/src/services/translation/response/spine-response-handler.ts

```diff
@@ -542,3 +533,51 @@ export class ReleaseResponseHandler extends SpineResponseHandler<hl7V3.Prescript
     }
   }
 }
+
+export class ReleaseRejectionHandler extends SpineResponseHandler<hl7V3.PrescriptionReleaseRejectRoot> {
+  protected extractErrorCodes(
+    sendMessagePayload: hl7V3.SendMessagePayload<hl7V3.PrescriptionReleaseRejectRoot>
+  ): Array<hl7V3.Code<string>> {
+    const errorCode = sendMessagePayload.ControlActEvent.subject.PrescriptionReleaseReject
+      .pertinentInformation?.pertinentRejectionReason?.value
+    return errorCode ? [errorCode] : []
+  }
+
+  protected handleErrorResponse(
+    sendMessagePayload: hl7V3.SendMessagePayload<hl7V3.PrescriptionReleaseRejectRoot>,
+    logger: pino.Logger
+  ): TranslatedSpineResponse {
+    const spineResponse = super.handleErrorResponse(sendMessagePayload, logger)
+    const operationOutcome = spineResponse.fhirResponse as fhir.OperationOutcome
+    operationOutcome.issue.forEach((issue) => {
+      if (ReleaseRejectionHandler.issueNeedsDiagnosticInfo(issue)) {
+        issue.diagnostics = ReleaseRejectionHandler.getDiagnosticInfo(sendMessagePayload)
+      }
+      return issue
+    })
+    return spineResponse
+  }
+
+  private static issueNeedsDiagnosticInfo(issue: fhir.OperationOutcomeIssue) {
+    const issueDetails = issue.details.coding
+    return issueDetails.some(issue => issue.code === "PRESCRIPTION_WITH_ANOTHER_DISPENSER")
```

Comment:
feel like we should have a const with the value `"PRESCRIPTION_WITH_ANOTHER_DISPENSER"` we could use here rather than the string?